src/service/overseas.entities.service.ts

```diff
@@ -11,18 +11,21 @@ export const createOverseasEntity = async (
   req: Request,
   session: Session,
   transactionId: string,
-  isSaveAndResumeFeatureActive: boolean = false
+  isSaveAndResumeFeatureActive: boolean = false,
+  data?: ApplicationData
 ): Promise<string> => {
 
   logger.infoRequest(req, `Calling 'postOverseasEntity' for transaction id '${transactionId}'`);
 
+  const appData: ApplicationData = typeof data !== "undefined" ? data : getApplicationData(session);
```

Comment:
If you don't need to check specifically for undefined you can maybe only check if data exists.
const appData: ApplicationData = !data ? getApplicationData(session) : data

This is just a suggestion in case you end up with null for some reason.

Reply:
I agree, the nullish coalescing operator is more suitable in this case.